### PR TITLE
firefox esr

### PR DESCRIPTION
**Pull request by zdtcd**
Built for:
- [x] x86_64 from x86_64 bootstrap - TESTED
- [x] x86_64-musl from x86_64-musl bootstrap - TESTED
- [x] i686 from i686 bootstrap - BUILT and smoke TESTED.
    - Setting search engine seems broken, the search engine in address bar survives one restart,
    but it's reset to Google on the second restart, the default search engine in setting,
    and search bar. and in homepage doesn't survive the restart.
    Reproducible with Firefox 69.0
    - add-on works fine
- [ ] i686 from x86_64 bootstrap - IN PROGRESS
    - ~tool-chain crashed~ I forget to remove `-march=native` in my configuration
    - depends on #14401 
- [x] aarch64-musl from x86_64-musl bootstrap - smoke TESTED
- [x] ppc64le-musl from x86_64-musl bootstrap

### CODE REVIEW

**Comment by Chocimier** [comment]
You can put script into files/ .

**Comment by Hoshpak** [comment]
Could you also test building for i686 and at least one supported cross architecture? These thinks tend to fail a lot so I would like to have it tested before merging.

**Comment by zdtcd** [comment]
On September 8, 2019 9:02:33 AM UTC, Helmut Pozimski <notifications@github.com> wrote:
>Could you also test building for i686 and at least one supported cross
>architecture? These thinks tend to fail a lot so I would like to have
>it tested before merging.

I'll try to build for i686, ppc and arch64.

It would take time since my build machine is not very fast.
-- 
Danh

File: srcpkgs/firefox-i18n/files/update-lang-list [hunk 1]
```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
```
Comment (Johnnynator) [inline srcpkgs/firefox-i18n/files/update-lang-list]: I don't think updates scripts like this should be in the void-packages repo, xtools would be more appropriate.

Comment (zdtcd) [reply srcpkgs/firefox-i18n/files/update-lang-list]: Originally, I intended to add it into xtools.
However, this script is only useful for firefox and firefox-esr,
and @Chocimier told it's better to include it here.

Anyway, it's in separated commit, I'll drop it out of void-packages tree
if we can gain the consensus.

Comment (Chocimier) [reply srcpkgs/firefox-i18n/files/update-lang-list]: My reason to include it here is that it is tied to one package, so
- easier to find
- xtools is useful for non-maintainers - we ask for xuname output for bug reports - and populating /bin with package-tied scripts for them isn't great
- we have similar scripts for bash, iana-etc.

Comment (Hoshpak) [reply srcpkgs/firefox-i18n/files/update-lang-list]: I agree with @Chocimier , having such scripts in the tree is highly useful. I've used the glibc update script in the past to update upstream patches which is also in tree.

**Comment by zdtcd** [comment]
Testing is stuck because of update on gcc and cross toolchain, hence xbps-src want to rebuild a lot of packages.
I'll get back to work on this pr when void buildbot settled.

**Comment by zdtcd** [comment]
This PR could be pulled for now.
There're 2 issues:
- Cross compile i686 from x86_64 depends on #14401 but most people compile i686 from i686 bootstrap
- cross compile ppc64le from x86_64 have some problems with cross-compiler toolchain

**Pull request merged**